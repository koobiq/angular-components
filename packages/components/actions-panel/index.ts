export * from './actions-panel';
export * from './actions-panel-config';
export * from './actions-panel-container';
export * from './actions-panel-ref';
export * from './module';
