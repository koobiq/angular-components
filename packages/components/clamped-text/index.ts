export * from './clamped-text';
export * from './module';
export * from './types';
