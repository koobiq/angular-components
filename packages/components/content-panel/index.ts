export * from './content-panel';
export * from './module';
export * from './resizable';
