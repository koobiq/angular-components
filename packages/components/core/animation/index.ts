export * from './animation';
export { fadeAnimation } from './fade-animations';
export { kbqSelectAnimations } from './select-animations';
