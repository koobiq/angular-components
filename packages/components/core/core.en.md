The `core` module is a foundational part of the **Koobiq** design system.  
It provides essential utilities, services, and components used across other modules in the system.
