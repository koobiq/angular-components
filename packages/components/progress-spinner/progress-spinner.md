`<kbq-progress-spinner>` - компонент, отображающий индикатор загрузки в виде спиннера.

### Simple progress spinner

Элемент `<kbq-progress-spinner>` может быть использован отдельно для создания сектора прогресса с темой Koobiq.

```html
<kbq-progress-spinner></kbq-progress-spinner>
```

<!-- example(progress-spinner-overview) -->

### Props

#### value

Степень заполнения сектора, определяющая прогресс, зависит от свойства value.
Его значение может меняться в диапазоне [0, 100].
Значение по-умолчанию: 0

Такое значение заполнит 30% сектора:

```html
<kbq-progress-spinner value="30"></kbq-progress-spinner>
```

#### mode

Возможные значения: 'determinate', 'indeterminate'.
Значение по-умолчанию: 'determinate'.

Для отображения неопределенного по завершенности прогресса:

```html
<kbq-progress-spinner [mode]="'indeterminate'"></kbq-progress-spinner>
```

Для отображения определенного по завершенности прогресса:

```html
<kbq-progress-spinner
    [mode]="'determinate'"
    value="30"
></kbq-progress-spinner>
```

<!-- example(progress-spinner-indeterminate) -->
