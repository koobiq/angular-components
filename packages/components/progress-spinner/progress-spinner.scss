@use '../core/styles/functions/timing';

@use '../core/styles/common/tokens' as *;

@use './progress-spinner-theme' as *;

@keyframes kbq-progress-spinner-indeterminate {
    100% {
        transform: rotateZ(270deg);
    }
}

.kbq-progress-spinner {
    display: flex;
    flex-direction: row;

    gap: kbq-css-variable(progress-spinner-size-compact-content-gap-horizontal);
}

.kbq-progress-spinner__circle {
    fill: none;
    stroke: black;
    stroke-dasharray: 295%;
    stroke-width: 13%;
    transition: stroke-dashoffset 0.3s;
    transform-origin: center center;
}

.kbq-progress-spinner__inner {
    width: kbq-css-variable(progress-spinner-size-compact-size);
    height: kbq-css-variable(progress-spinner-size-compact-size);

    transform: rotateZ(-90deg);

    font-size: 0;
}

.kbq-progress-spinner__content {
    .kbq-progress-spinner-text {
        margin-bottom: kbq-css-variable(progress-spinner-size-compact-content-gap-vertical);
    }
}

.kbq-progress-spinner_big {
    gap: kbq-css-variable(progress-spinner-size-big-content-gap-horizontal);

    & .kbq-progress-spinner__inner {
        width: kbq-css-variable(progress-spinner-size-big-size);
        height: kbq-css-variable(progress-spinner-size-big-size);
    }

    & .kbq-progress-spinner__circle {
        stroke-width: 6%;
    }
}

.kbq-progress-spinner_indeterminate .kbq-progress-spinner__inner {
    animation: kbq-progress-spinner-indeterminate 1.5s timing.$ease-in-out-quad infinite;

    // TODO: rework this place
    .kbq-progress-spinner__circle {
        stroke-dashoffset: 80%;
    }
}

@include kbq-progress-spinner-theme();
@include kbq-progress-spinner-typography();
