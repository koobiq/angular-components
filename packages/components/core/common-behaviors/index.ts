import { InjectionToken } from '@angular/core';

export { CanColor, CanColorCtor, KbqComponentColors, ThemePalette, mixinColor } from './color';
export { KBQ_SANITY_CHECKS, KbqCommonModule } from './common-module';
export { CanDisable, CanDisableCtor, mixinDisabled } from './disabled';
export { CanUpdateErrorState, CanUpdateErrorStateCtor, mixinErrorState } from './error-state';
export { HasTabIndex, HasTabIndexCtor, mixinTabIndex } from './tabindex';

export const KBQ_PARENT_ANIMATION_COMPONENT = new InjectionToken<any>('kbq-parent-animation-component');
