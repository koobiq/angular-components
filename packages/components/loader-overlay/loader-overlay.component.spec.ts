import { Component, ViewChild } from '@angular/core';
import { TestBed } from '@angular/core/testing';
import { BrowserModule, By } from '@angular/platform-browser';
import { ThemePalette } from '@koobiq/components/core';
import { KbqProgressSpinnerModule } from '@koobiq/components/progress-spinner';
import {
    KbqLoaderOverlay,
    KbqLoaderOverlayCaption,
    KbqLoaderOverlayIndicator,
    KbqLoaderOverlayModule,
    KbqLoaderOverlayText
} from './index';

describe('KbqLoaderOverlay', () => {
    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                BrowserModule,
                KbqProgressSpinnerModule,
                KbqLoaderOverlayModule
            ],
            declarations: [
                OverlayWithParams,
                OverlayNoParams,
                OverlayWithExternalParams
            ]
        }).compileComponents();
    });

    it('should be transparent by default', () => {
        const fixture = TestBed.createComponent(OverlayWithParams);
        fixture.detectChanges();

        expect(fixture.componentInstance.overlay.transparent).toBeTruthy();
    });

    it('should render params', () => {
        const fixture = TestBed.createComponent(OverlayWithParams);
        fixture.detectChanges();

        const indicatorElement = fixture.debugElement.query(By.css('kbq-progress-spinner'));
        const textElement = fixture.debugElement.query(By.css('.kbq-loader-overlay-text'));
        const captionElement = fixture.debugElement.query(By.css('.kbq-loader-overlay-caption'));

        expect(indicatorElement).toBeDefined();
        expect(textElement.nativeElement.textContent.trim()).toContain('Создание отчета');
        expect(captionElement.nativeElement.textContent.trim()).toBe('18,7 МБ из 25 МБ — осталось 2 мин');
    });

    it('should render only spinner with modifier', () => {
        const fixture = TestBed.createComponent(OverlayNoParams);
        fixture.detectChanges();

        const indicatorElement = fixture.debugElement.query(By.css('kbq-progress-spinner'));
        const containerElement = fixture.debugElement.query(By.css('kbq-loader-overlay'));
        const textElement = fixture.debugElement.query(By.css('.kbq-loader-overlay-text'));
        const captionElement = fixture.debugElement.query(By.css('.kbq-loader-overlay-caption'));

        expect(indicatorElement).toBeDefined();
        expect(containerElement.nativeElement.classList).toContain('kbq-loader-overlay_empty');
        expect(textElement).toBeNull();
        expect(captionElement).toBeNull();
    });

    it('should render external params', () => {
        const fixture = TestBed.createComponent(OverlayWithExternalParams);
        fixture.detectChanges();

        const indicatorElement = fixture.debugElement.query(By.directive(KbqLoaderOverlayIndicator));
        const textElement = fixture.debugElement.query(By.directive(KbqLoaderOverlayText));
        const captionElement = fixture.debugElement.query(By.directive(KbqLoaderOverlayCaption));

        expect(indicatorElement.nativeElement.classList).toContain('kbq-error');
        expect(textElement.nativeElement.textContent.trim()).toBe('Создание отчета');
        expect(captionElement.nativeElement.textContent.trim()).toBe('18,7 МБ из 25 МБ — осталось 2 мин');
    });
});

@Component({
    selector: 'overlay-with-params',
    template: `
        <div class="loader-overlay-container">
            text text text text text text text text text text text text text text text text text text text text

            <kbq-loader-overlay [text]="'Создание отчета'" [caption]="'18,7 МБ из 25 МБ — осталось 2 мин'" />
        </div>
    `
})
class OverlayWithParams {
    @ViewChild(KbqLoaderOverlay) overlay: KbqLoaderOverlay;
}

@Component({
    selector: 'overlay-no-params',
    template: `
        <div class="loader-overlay-container">
            text text text text text text text text text text text text text text text text text text text text

            <kbq-loader-overlay />
        </div>
    `
})
class OverlayNoParams {}

@Component({
    selector: 'overlay-with-external-params',
    template: `
        <div class="loader-overlay-container">
            text text text text text text text text text text text text text text text text text text text text

            <kbq-loader-overlay>
                <kbq-progress-spinner
                    [mode]="'indeterminate'"
                    [color]="themePalette.Error"
                    kbq-loader-overlay-indicator
                />

                <div kbq-loader-overlay-text>Создание отчета</div>
                <div kbq-loader-overlay-caption>18,7 МБ из 25 МБ — осталось 2 мин</div>
            </kbq-loader-overlay>
        </div>
    `
})
class OverlayWithExternalParams {
    themePalette = ThemePalette;
}
