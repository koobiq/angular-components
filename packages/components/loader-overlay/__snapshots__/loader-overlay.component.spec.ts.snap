// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KbqLoaderOverlay should assign default css classes including size-related css-class 1`] = `
{
  "kbq-loader-overlay": true,
  "kbq-loader-overlay_big": true,
  "kbq-loader-overlay_transparent": true,
}
`;
